src/lib/components/api.js

```diff
@@ -0,0 +1,164 @@
+  for (let pr of pullRequests) {
+    for (let file of pr.files.nodes) {
+      if (
+        !filesToRetrieve.some(item => item.path === file.path) &&
+        (file.path.substring(file.path.length - 3) === ".md" ||
+          file.path.substring(file.path.length - 9) === ".markdown") &&
+        file.path.substring(0, 6) === "rules/"
+      ) {
+        filesToRetrieve = [...filesToRetrieve, {
+          path: file.path,
+          author: pr.author.login,
+            timestamp: new Date(pr.mergedAt),
+          },
+        ];
+      }
+    }
+  }
+  return filesToRetrieve;
+}
+
+export async function requestPullRequests(numberOfRules, author, token) {
+  var response = await fetch(apiBaseUrl, {
+    method: "POST",
+    headers: {
+      Authorization: `bearer ${token}`,
+    },
+    body: JSON.stringify({
+      query: `{
+			search(query: "repo:${githubOwner}/${githubRepo} is:pr base:main is:merged sort:updated-desc ${author ? "author:" + author : ""
+        }", type: ISSUE, first: ${numberOfRules + 10}) {
+			nodes {
+				... on PullRequest {
+				author {
+					login
+				}
+				files(first: 10) {
+					nodes {
+					path
+					}
+				}
+				mergedAt
+				}
+			}
+			}
+		}`,
+    }),
+  })
+    .then((res) => res.json())
+    .catch((error) => { return error });
+
+  return response.data.search.nodes || null;
+}
+
+async function requestMultipleFileContents(list, numberOfRules, token) {
+  var promises = [];
+  for (var i = 0; i < numberOfRules; i++) {
+    promises.push(
+      fetch(apiBaseUrl, {
+        method: "POST",
+        headers: {
+          Authorization: `bearer ${token}`,
+        },
+        body: JSON.stringify({
+          query: `{
+				repository(name: "${githubRepo}", owner: "${githubOwner}") {
+				object(expression: "main:${list[i].path}") {
+					... on Blob {
+					text
+					}
+				}
+				}
+			}`,
+        }),
+      })
+    );
+  }
+
+  var contents = [];
+  await Promise.all(promises)
+    .then((returnedPromises) => Promise.all(returnedPromises.map((res) => res.json())))
+    .then((returnedObjects) =>
+      returnedObjects.forEach((obj) => {
+        if (obj.data.repository.object != null) {
+          contents = [...contents, obj.data.repository.object.text];
+        } else {
+          contents = [...contents, null];
+        }
+      })
+    )
+    .catch((error) => { return error });
+
+  return contents || null;
+}
+
+async function requestSingleFileContents(file, token) {
+  var response = await fetch(apiBaseUrl, {
+    method: "POST",
+    headers: {
+      Authorization: `bearer ${token}`,
+    },
+    body: JSON.stringify({
+      query: `{
+		rateLimit {
+			remaining
+		}
+		repository(name: "${githubRepo}", owner: "${githubOwner}") {
+			object(expression: "main:${file}") {
+			... on Blob {
+				text
+			}
+			}
+		}
+		}`,
+    }),
+  })
+    .then((res) => res.json())
+    .catch((error) => { return error });
+
+  return response.data.repository.object.text || null;
```

Comment:
These lines where we do || null, why do we have to do that? It's already evaluating to falsy, so we shouldn't need to set it to a specific falsy value should we?